presto-hive/src/main/java/com/facebook/presto/hive/HiveWriter.java

```diff
@@ -82,10 +82,11 @@ public void append(Page dataPage)
         inputSizeInBytes += dataPage.getSizeInBytes();
     }
 
-    public void commit()
+    public Optional<Page> commit()
     {
-        fileWriter.commit();
+        Optional<Page> fileStatistics = fileWriter.commit();
```

Comment:
IIRC, each commit or each HiveWriter can only create one file right? @wenleix, is that the case?

Reply:
Yes correct. One file per HiveWriter.

Reply:
Correct. We create separate `HiveWriter` for each file.